src/Microsoft.AspNet.OData.Shared/Builder/SingletonConfiguration.cs

```diff
@@ -41,5 +41,14 @@ public SingletonConfiguration(ODataModelBuilder modelBuilder, EntityTypeConfigur
             : base(modelBuilder, entityType, name)
         {
         }
+
+        /// <summary>
+        /// Adds TSubClass to the list of derived type constraints.
+        /// </summary>
+        public SingletonConfiguration AddDerivedTypeConstraint<TSubClass>() where TSubClass : class
```

Comment:
>AddDerivedTypeConstraint<TSubClass>() [](start = 38, length = 37)

For design perspective, add a non-generic method to support adding TypeConstraint.

Call the non-generic method in the generic method.

Reply:
Same for others.

---
In reply to: [358416790](https://github.com/OData/WebApi/pull/1977#discussion_r358416790) [](ancestors = 358416790)

Reply:
somthing likes `public SingletonConfiguration AddDerivedTypeConstraint(params Type type)
{}`

---
In reply to: [358416861](https://github.com/OData/WebApi/pull/1977#discussion_r358416861) [](ancestors = 358416861,358416790)